GirafRest.Test/Controllers/UserControllerTest.cs

```diff
@@ -1024,9 +1121,15 @@ public void UpdateUserSettings__AuthenticationChecks(int authUser, int userToEdi
             _testContext.MockUserManager.MockLoginAsUser(user);
             var dto = UserSettings[0];
 
-            var res = usercontroller.UpdateUserSettings(_testContext.MockUsers[userToEdit].Id, dto).Result;
+            var res = usercontroller.UpdateUserSettings(_testContext.MockUsers[userToEdit].Id, dto).Result as ObjectResult;
 
-            Assert.Equal(expectedError, res.ErrorCode);
+            if (expectedError != ErrorCode.NoError) {

```

Comment:
Again: Should be two tests instead of one with an `if` block.

recall that a unit test should clearly specify it's expected result in the method header, so a new dev can immediatly see if it should assert an error or a success, etc.